st3/sublime_lib/syntax.py

```diff
@@ -1,38 +1,52 @@
 import sublime
 
 import re
+from collections import namedtuple
 
 __all__ = ['list_syntaxes', 'get_syntax_for_scope']
 
-class SyntaxInfo():
-    __slots__ = [ 'path', 'name', 'scope', 'hidden' ]
-    
-    def __init__(self, path, name=None, scope=None, hidden=False):
-        self.path = path
-        self.name = name
-        self.scope = scope
-        self.hidden = hidden
+
+SyntaxInfo = namedtuple('SyntaxInfo', ['path', 'name', 'scope', 'hidden'])
+SyntaxInfo.__new__.__defaults__ = (None, None, False)  # type: ignore
```

Comment:
This looks good; I actually started with `namedtuple` and stopped using it because I didn't know about `__defaults__`. Just to be sure, this is for sure a stable feature?

Reply:
Yes, it's referenced on https://docs.python.org/3/reference/datamodel.html.